pkg/apply/task/inv_set_task.go

```diff
@@ -124,3 +128,36 @@ func (i *InvSetTask) Cancel(_ *taskrunner.TaskContext) {}
 
 // StatusUpdate is not supported by the InvSetTask.
 func (i *InvSetTask) StatusUpdate(_ *taskrunner.TaskContext, _ object.ObjMetadata) {}
+
+func (i InvSetTask) getApplyReoncileStatus(taskContext *taskrunner.TaskContext) map[object.ObjMetadata]status.ApplyReconcileStatus {
+	applyStatus := map[object.ObjMetadata]status.ApplyReconcileStatus{}
+	// capture the apply status
+	for _, obj := range taskContext.SuccessfulApplies() {
+		applyStatus[obj] = status.ApplyReconcileStatus{
+			ApplyStatus: status.ApplySucceeded,
+		}
+	}
+	for _, obj := range i.PrevInventory.Intersection(taskContext.FailedApplies()) {
+		applyStatus[obj] = status.ApplyReconcileStatus{
+			ApplyStatus: status.ApplyFailed,
+		}
+	}
+	for _, obj := range i.PrevInventory.Intersection(taskContext.SkippedApplies()) {
+		applyStatus[obj] = status.ApplyReconcileStatus{
+			ApplyStatus: status.ApplySkipped,
+		}
+	}
+	for _, obj := range i.PrevInventory.Intersection(taskContext.FailedDeletes()) {
+		applyStatus[obj] = status.ApplyReconcileStatus{
+			ApplyStatus: status.PruneFailed,
+		}
+	}
+	for _, obj := range i.PrevInventory.Intersection(taskContext.SkippedDeletes()) {
```

Comment:
Does this cover all the apply statuses?

Reply:
Yes, it covers all the apply statuses. The reconcile status are to be added later.